File: rest-spring-cxf-shiro/src/main/webapp/resources/js/1/product_service.js
var ProductService = function() {
    var BASE = '/rest-spring-cxf-shiro/ws/rest';
    return {
        retrieveAllProducts: function(callback) {
            $.ajax({
                type: 'get',
                url: BASE + '/products',
                dataType: 'json',
                success: function(productList) {
                    ProductView.renderProductTable(productList);
                    callback();
                }
            });
        },
        retrieveProductById: function(id) {
            $.ajax({
                type: 'get',
                url: BASE + '/product/' + id,
                dataType: 'json',
                success: function(product) {
                    var title = 'Edit Product';
                    ProductView.renderProductModal(title, product);
                }
            });
        },
        retrieveProductsByName: function(name) {
            $.ajax({
                type: 'post',
                url: BASE + '/products',
                data: {
                    name: name
                },
                dataType: 'json',
                contentType: 'application/x-www-form-urlencoded',
                success: function(productList) {
                    ProductView.renderProductTable(productList);
                }
            });
        },
        createProduct: function(product) {
            $.ajax({
                type: 'post',
                url: BASE + '/product',
                data: JSON.stringify(product),
                dataType: 'json',
                contentType: 'application/json',
                success: function(product) {
                    ProductView.insertProductRow(product);
                }
            });
        },
        updateProductById: function(id, product) {
            $.ajax({
                type: 'put',
                url: BASE + '/product/' + id,
                data: JSON.stringify(product),
                dataType: 'json',
                contentType: 'application/json',
                success: function(product) {
                    ProductView.updateProductRow(product);
                }
            });
        },
        deleteProductById: function(id) {
            $.ajax({
                type: 'delete',
                url: BASE + '/product/' + id,
                dataType: 'json',
                success: function() {
                    ProductView.deleteProductRow(id);
                }
            });
        }
    };
}();

File: rest-cxf/src/main/java/me/czd/ws/rest_cxf/ProductService.java
package me.czd.ws.rest_cxf;

import java.util.List;
import java.util.Map;

import javax.ws.rs.Consumes;
import javax.ws.rs.DELETE;
import javax.ws.rs.FormParam;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

/**
 * rest 服务的接口， 套用别人的名词，服务就是提供方，就是生产方 这里使用了java RS 的规范，毕竟java 的 标准规范
 * 
 * retrieve 取回 程序员英语也不能落后
 * 
 * 这里可以说 是关于 Product 的一套操作
 * 
 * @author Administrator
 *
 */
public interface ProductService {

	//获取所有 product 用复数 uri
	@GET
	@Path("/products")
	@Produces(MediaType.APPLICATION_JSON)
	List<Product> retrieveAllProducts();

	//通过id 来获取 product
	@GET
	@Path("/product/{id}")
	@Produces(MediaType.APPLICATION_JSON)
	Product retrieveProductById(@PathParam("id") long id);

	//通过表单提交来 根据name 获取products
	@POST
	@Path("/products")
	@Consumes(MediaType.APPLICATION_FORM_URLENCODED)
	@Produces(MediaType.APPLICATION_JSON)
	List<Product> retrieveProductsByName(@FormParam("name") String name);

	//通过 post 来创建 product
	@POST
	@Path("/product")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	Product createProduct(Product product);

	//通过 put 来进行 某个id 的更新
	@PUT
	@Path("/product/{id}")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	Product updateProductById(@PathParam("id") long id, Map<String, Object> fieldMap);

	//通过 delete 来进行 某个id 的删除
	@DELETE
	@Path("/product/{id}")
	@Produces(MediaType.APPLICATION_JSON)
	Product deleteProductById(@PathParam("id") long id);

}
